conference/debug_panel.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+
+"""
+Things in this file are related to a special debug panel that helps us debug
+things in production.
+"""
+
+import platform
+import subprocess
+
+import django
+from django.contrib.admin.views.decorators import staff_member_required
+from django.conf import settings
+from django.template.response import TemplateResponse
+
+
+def get_current_commit_hash():
+    command = 'git rev-parse HEAD'
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, cwd=settings.PROJECT_DIR
+    )
+    full_hash = process.communicate()[0]
+    return "%s %s" % (full_hash[:8], full_hash)
+
+
+commit_hash_in_current_process = get_current_commit_hash()
+
+
+@staff_member_required
+def debug_panel(request):
+
+    debug_vars = [
+        ('Current_Commit_Hash', get_current_commit_hash()),
```

Comment:
Why do you need this information twice?

Reply:
because the first one is executing every time we reload the page, and second keeps the value from when the process was launched (basically it's filled on django's start when it populates the URLs) they can, in edge cases, provide a different results – for example if you do `pull` or `checkout` but do not restart the process the hashes will be different.